application/templates/cronjob.yaml

```
@@ -54,13 +54,7 @@ spec:
           annotations: {{ toYaml . | nindent 12 }}
           {{- end }}
         spec:
-          {{- if $.Values.rbac.enabled }}
-          {{- if $.Values.rbac.serviceAccount.name }}
-          serviceAccountName: {{ $.Values.rbac.serviceAccount.name }}
-            {{- else }}
-          serviceAccountName: {{ template "application.name" $ }}
-          {{- end }}
-          {{- end }}
+          serviceAccountName: {{ include "application.serviceAccountName" $ }}
```

This may be another feature, but personally, I think we should add this:

```suggestion
          automountServiceAccountToken: {{ $.Values.rbac.enabled }}
          serviceAccountName: {{ include "application.serviceAccountName" $ }}
```

If RBAC is disabled, we shouldn't even allow mounting tokens for the default service account, I'd say. What do you think?

---

Sounds good to me,  that seems like a reasonable assumption.